packages/blockchain/src/blockchain.ts

```diff
@@ -529,7 +529,9 @@ export class Blockchain implements BlockchainInterface {
         let dbOps: DBOp[] = []
 
         if (block.common.chainId() !== this.common.chainId()) {
-          throw new Error('Chain mismatch while trying to put block or header')
+          throw new Error(
+            `Chain mismatch while trying to put block or header. Block chain ID: ${block.common.chainId}, blockchain chain ID: ${this.common.chainId}`
```

Comment:
Very small nit, but `Block chain ID` and `blockchain chain ID` can be confusing and sound like the same thing.
```suggestion
            `Chain mismatch while trying to put block or header. Chain ID of block: ${block.common.chainId}, chain ID of blockchain : ${this.common.chainId}`
```